
# Class: analysis


An Analysis is a data transformation that transforms input data to output data. The workflow used to achieve this transformation and the individual steps are also captured.

URI: [https://w3id.org/GHGA-Submission-Metadata-Schema/Analysis](https://w3id.org/GHGA-Submission-Metadata-Schema/Analysis)


[![img](https://yuml.me/diagram/nofunky;dir:TB/class/[Submission],[Study],[File],[EgaAccessionMixin],[Dataset],[DataTransformation],[File]<has%20output%201..*-++[Analysis&#124;type:string%20%3F;reference_genome:string;reference_chromosome:string;description:string%20%3F;alias:string%20%3F;accession:string;ega_accession:string;title(i):string%20%3F;id(i):string;xref(i):string%20*],[Study]<has%20study%201..1-++[Analysis],[File]<has%20input%201..*-++[Analysis],[Dataset]++-%20has%20analysis%200..*>[Analysis],[Dataset]-%20has%20analysis(i)%200..1>[Analysis],[Submission]-%20has%20analysis(i)%200..1>[Analysis],[Submission]++-%20has%20analysis%200..*>[Analysis],[Analysis]uses%20-.->[AccessionMixin],[Analysis]uses%20-.->[EgaAccessionMixin],[DataTransformation]^-[Analysis],[AccessionMixin])](https://yuml.me/diagram/nofunky;dir:TB/class/[Submission],[Study],[File],[EgaAccessionMixin],[Dataset],[DataTransformation],[File]<has%20output%201..*-++[Analysis&#124;type:string%20%3F;reference_genome:string;reference_chromosome:string;description:string%20%3F;alias:string%20%3F;accession:string;ega_accession:string;title(i):string%20%3F;id(i):string;xref(i):string%20*],[Study]<has%20study%201..1-++[Analysis],[File]<has%20input%201..*-++[Analysis],[Dataset]++-%20has%20analysis%200..*>[Analysis],[Dataset]-%20has%20analysis(i)%200..1>[Analysis],[Submission]-%20has%20analysis(i)%200..1>[Analysis],[Submission]++-%20has%20analysis%200..*>[Analysis],[Analysis]uses%20-.->[AccessionMixin],[Analysis]uses%20-.->[EgaAccessionMixin],[DataTransformation]^-[Analysis],[AccessionMixin])

## Parents

 *  is_a: [DataTransformation](DataTransformation.md) - A data transformation technique used to analyze and interpret data to gain a better understanding of it.

## Uses Mixin

 *  mixin: [AccessionMixin](AccessionMixin.md) - Mixin for entities that can be assigned a GHGA accession.
 *  mixin: [EgaAccessionMixin](EgaAccessionMixin.md) - Mixin for entities that can be assigned an EGA accession, in addition to GHGA accession.

## Referenced by Class

 *  **[Dataset](Dataset.md)** *[dataset➞has analysis](dataset_has_analysis.md)*  <sub>0..\*</sub>  **[Analysis](Analysis.md)**
 *  **None** *[has analysis](has_analysis.md)*  <sub>0..1</sub>  **[Analysis](Analysis.md)**
 *  **[Submission](Submission.md)** *[submission➞has analysis](submission_has_analysis.md)*  <sub>0..\*</sub>  **[Analysis](Analysis.md)**

## Attributes


### Own

 * [analysis➞type](analysis_type.md)  <sub>0..1</sub>
     * Description: The type of the Analysis. Either Reference Alignment (BAM) or Sequence Variation (VCF)
     * Range: [String](types/String.md)
 * [analysis➞reference genome](analysis_reference_genome.md)  <sub>1..1</sub>
     * Description: A published genetic sequence that is used as a reference sequence against which other sequences are compared. Reference genome(s) or annotation(s) used for prior analyses (eg: GRCh38.p13).
     * Range: [String](types/String.md)
 * [analysis➞reference chromosome](analysis_reference_chromosome.md)  <sub>1..1</sub>
     * Description: The reference chromosome used for this Analysis.
     * Range: [String](types/String.md)
 * [analysis➞has input](analysis_has_input.md)  <sub>1..\*</sub>
     * Description: The input data File entities used in the Analysis.
     * Range: [File](File.md)
     * in subsets: (restricted)
 * [analysis➞has study](analysis_has_study.md)  <sub>1..1</sub>
     * Description: The Study entity associated with this Analysis.
     * Range: [Study](Study.md)
 * [analysis➞has output](analysis_has_output.md)  <sub>1..\*</sub>
     * Description: The output data File entities generated by this Analysis.
     * Range: [File](File.md)
     * in subsets: (restricted)
 * [analysis➞description](analysis_description.md)  <sub>0..1</sub>
     * Description: Describing how an Analysis was carried out. (e.g.: computational tools, settings, etc.).
     * Range: [String](types/String.md)
 * [analysis➞alias](analysis_alias.md)  <sub>0..1</sub>
     * Description: An alias uniquely identifying this Analysis entitiy.
     * Range: [String](types/String.md)
     * in subsets: (restricted)

### Inherited from data transformation:

 * [named thing➞id](named_thing_id.md)  <sub>1..1</sub>
     * Description: The internal unique identifier for an entity.
     * Range: [String](types/String.md)
     * in subsets: (restricted)
 * [named thing➞xref](named_thing_xref.md)  <sub>0..\*</sub>
     * Description: Holds one or more database cross references for an entity.
     * Range: [String](types/String.md)
 * [title](title.md)  <sub>0..1</sub>
     * Description: The title that describes an entity.
     * Range: [String](types/String.md)

### Mixed in from accession mixin:

 * [accession mixin➞accession](accession_mixin_accession.md)  <sub>1..1</sub>
     * Description: A unique GHGA identifier assigned to an entity for the sole purpose of referring to that entity in a global scope.
     * Range: [String](types/String.md)

### Mixed in from ega accession mixin:

 * [ega accession mixin➞ega accession](ega_accession_mixin_ega_accession.md)  <sub>1..1</sub>
     * Description: A unique European Genome-Phenome Archive (EGA) identifier assigned to an entity for the sole purpose of referring to that entity within the EGA federated network.
     * Range: [String](types/String.md)

## Other properties

|  |  |  |
| --- | --- | --- |
| **Aliases:** | | data analysis |

